### PR TITLE
NNX from_pretrained Deep-Merge & tid2eid Routing Table Sideloading

### PR DESCRIPTION
**Pull request by AusarYao28**
# Description

forward KL divergence exceeded 4.8. Inspecting model_creation_utils.py revealed that raw_state_dict.update(restored_params) performed a shallow top-level overwrite of raw_state_dict["base"], wiping non-params NNX variable collections (Tid2EidVar). Furthermore, to_maxtext.py only converts nnx.Param weights, leaving the (129280, 58) token-to-expert table tid2eid at all zeros (routing every token to Expert 0).

FIXES: b/562177048
FIXES: b/563048036

# Tests

e2e test https://cloudlogging.app.goo.gl/8sf4ZKjS691ypzP19

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Comment by snehalv2002** [comment]
this is redundant since we merged #5226. Can close
